events/codepipeline_job.go

```diff
@@ -0,0 +1,74 @@
+// Copyright 2017 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+
+package events
+
+// CodePipelineEvent contains data from an event sent from AWS Codepipeline
+type CodePipelineEvent struct {
+	CodePipelineJob CodePipelineJob `json:"CodePipeline.job"`
+}
+
+// CodePipelineJob represents a job from an AWS CodePipeline event
+type CodePipelineJob struct {
+	ID        string           `json:"id"`
+	AccountID string           `json:"accountId"`
+	Data      CodePipelineData `json:"data"`
+}
+
+// CodePipelineData represents a job from an AWS CodePipeline event
+type CodePipelineData struct {
+	ActionConfiguration CodePipelineActionConfiguration `json:"actionConfiguration"`
+	InputArtifacts      []CodePipelineInputArtifact     `json:"inputArtifacts"`
+	OutPutArtifacts     []CodePipelineOutputArtifact    `json:"outputArtifacts"`
+	ArtifactCredentials CodePipelineArtifactCredentials `json:"artifactCredentials"`
+	ContinuationToken   string                          `json:"continuationToken"`
+}
+
+// CodePipelineActionConfiguration represents an Action Configuration
+type CodePipelineActionConfiguration struct {
+	Configuration CodePipelineConfiguration `json:"configuration"`
+}
+
+// CodePipelineConfiguration represents a configuration for an Action Configuration
+type CodePipelineConfiguration struct {
+	FunctionName   string `json:"FunctionName"`
+	UserParameters string `json:"UserParameters"`
+}
+
+// CodePipelineInputArtifact represents an input artifact
+type CodePipelineInputArtifact struct {
+	Location CodePipelineInputLocation `json:"location"`
+	Revision string                    `json:"revision"`
```

Comment:
regarding your comment on #48 
> In the unmarshal and marshal event, go converts this to " "

You can make the documented sample from https://docs.aws.amazon.com/codepipeline/latest/userguide/actions-invoke-lambda-function.html#actions-invoke-lambda-function-json-event-example work if you change this field's type to be pointer. `Revision *string`

Reply:
Changing `Revision string`  to `Revision *string` still results in Revision being converted to   `revision":" " ` if `"revision": null,` is given in the test doc. This causes tests to fail if using https://docs.aws.amazon.com/codepipeline/latest/userguide/actions-invoke-lambda-function.html#actions-invoke-lambda-function-json-event-example as a test doc.  Should I just replace `"revision": null,` with `"revision": "foo"` in the test doc?

Reply:
From https://play.golang.org/p/azCGae4c6jL, the type change ought to work.